python/lsst/daf/butler/core/schema.py

```diff
@@ -75,17 +77,19 @@ class SchemaBuilder:
     metadata : `sqlalchemy.MetaData`
         The sqlalchemy schema description.
     tables : `dict`
-        All created tables.
-    views : `dict`
-        All created views.
+        A mapping from table or view name to the associated SQLAlchemy object.
```

Comment:
I hadn't really noticed before that Schema requires SQLAlchemy even if Registry is implemented in non-SQL manner.

Reply:
Yeah, very much so.  In fact, while doing some work on DM-15210 it occurred to me that @pschella may have based some of the structure of `schema.yaml` on what SQLAlchemy wants - I've always found our description of foreign keys hard to parse, for instance, but it turns out they're quite convenient to pass to SQLAlchemy.  Anyhow, this bothers me too, but I'm not sure what to do about it and this ticket definitely isn't the place for it.

Reply:
The reason why I deliberately stayed pretty close to a one-to-one mapping between the YAML description and SQLAlchemy's object representation was that I expected us to have to use more complicated schema properties at some later stage. I thus assumed SQLAlchemy did what it did for a reason and it would make our lives easier in matching its expressiveness by having a one-to-one mapping as opposed to something that is perhaps a bit easier to read for the simple cases but requires ugly hacks for the complex ones.  This expectation may since have been proven wrong of course ;)